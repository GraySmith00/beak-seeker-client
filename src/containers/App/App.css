.app {
  margin: 0 auto;
  max-width: 375px;
  height: 100vh; }
