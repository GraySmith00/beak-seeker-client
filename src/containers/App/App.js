import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Home from '../Home/Home';
import SignIn from '../SignIn/SignIn';
import './App.css';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';

export class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <main>
            <Route
              exact
              path="/"
              render={() => {
                if (!this.props.currentUser) {
                  return <SignIn />;
                } else {
                  return <Home />;
                }
              }}
            />
            <Route path="/home" component={Home} />
          </main>
        </div>
      </Router>
    );
  }
}

App.propTypes = {
  currentUser: PropTypes.object
};

const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export default connect(mapStateToProps)(App);
