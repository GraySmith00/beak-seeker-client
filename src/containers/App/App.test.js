import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
