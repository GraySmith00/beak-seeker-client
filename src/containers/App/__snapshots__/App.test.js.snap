// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `"<div class=\\"App\\"><main><div class=\\"home\\"><nav><a href=\\"/hotspots\\">Nearby Hotspots</a><a href=\\"/leaderboard\\">Leaderboard</a></nav><main></main></div></main></div>"`;
